src/core/providers/qgsprovidersublayerdetails.h

```diff
@@ -72,6 +73,43 @@ class CORE_EXPORT QgsProviderSublayerDetails
      */
     void setType( QgsMapLayerType type ) { mType = type; }
 
+    /**
+     * Returns the layer's URI.
+     *
+     * \see setUri()
+     */
+    QString uri() const { return mUri; }
+
+    /**
+     * Sets the layer's \a uri.
+     *
+     * \see uri()
+     */
+    void setUri( const QString &uri ) { mUri = uri; }
+
+    /**
+     * Setting options for loading layers.
+     */
+    struct LayerOptions
+    {
+
+      /**
+       * Constructor for LayerOptions with \a transformContext.
+       */
+      explicit LayerOptions( const QgsCoordinateTransformContext &transformContext )
+        : transformContext( transformContext )
+      {}
+
+      QgsCoordinateTransformContext transformContext;
+    };
+
+    /**
+     * Creates a new QgsMapLayer object associated with the sublayer.
+     *
+     * Caller takes ownership of the returned layer.
```

Comment:
wouldn't std::unique_ptr<QgsMapLayer> be more idiomatic for that ? (but I suspect this may get in the way of SIP)

Reply:
Yeah, I've tried and failed to get a unique_ptr to work with sip. Unfortunately we're stuck with this till sip itself gains support...